assets/js/plugin-check-admin.js

```diff
@@ -172,14 +172,50 @@
 	 * @param {Object} data The response data.
 	 */
 	async function runChecks( data ) {
+		let isSuccessMessage = true;
 		for ( let i = 0; i < data.checks.length; i++ ) {
 			try {
 				const results = await runCheck( data.plugin, data.checks[ i ] );
+				const errorsLength = Object.values( results.errors ).length;
+				const warningsLength = Object.values( results.errors ).length;
+				if (
+					isSuccessMessage &&
+					( errorsLength > 0 || warningsLength > 0 )
+				) {
+					isSuccessMessage = false;
+				}
 				renderResults( results );
 			} catch ( e ) {
 				// Ignore for now.
 			}
 		}
+
+		renderResultsMessage( isSuccessMessage );
+	}
+
+	/**
+	 * Renders result message.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {boolean} isSuccessMessage Whether the message is a success message.
+	 */
+	function renderResultsMessage( isSuccessMessage ) {
+		let messageType = 'success';
+		let messageText = pluginCheck.successMessage;
+
+		if ( ! isSuccessMessage ) {
+			messageType = 'error';
+			messageText = pluginCheck.errorMessage;
+		}
```

Comment:
```suggestion
		const messageType = isSuccessMessage ? 'success' : 'error';
		const messageText = isSuccessMessage ? pluginCheck.successMessage : pluginCheck.errorMessage;
```

Const are a little better for performance of js. Maybe this is less readable.